api/src/main/java/org/apache/iceberg/OverwriteFiles.java

```diff
@@ -145,4 +151,50 @@
    */
   @Deprecated
   OverwriteFiles validateNoConflictingAppends(Long readSnapshotId, Expression conflictDetectionFilter);
+
+  /**
+   * Sets a conflict detection filter used to validate concurrently added data and delete files.
```

Comment:
Am I right to understand that we use rowFilter to do the detection, if this is not set?  Should we mention it?